libs/domain-core/src/lib/aggregate-repository.spec.ts

```diff
@@ -1,7 +1,98 @@
+/* eslint-disable max-classes-per-file */
+import { AggregateRepository } from './aggregate-repository';
+import { Event } from './event';
+import { EventStore } from './event-store';
+import { TestAggregate, TestEvent } from './aggregate.spec';
+
+class TestEventStore implements EventStore {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars,class-methods-use-this
+  append(stream: string, events: Event[], expectedRevision: number): Promise<void> {
+    return Promise.resolve(undefined);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars,class-methods-use-this
+  read(stream: string): Promise<Event[]> {
+    return Promise.resolve([]);
+  }
+}
+
+class TestAggregateRepository extends AggregateRepository<TestAggregate> {
+  // eslint-disable-next-line @typescript-eslint/no-useless-constructor
+  constructor(eventStore: EventStore) {
+    super(eventStore);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getNewInstance(): TestAggregate {
+    return new TestAggregate();
+  }
+}
+
 describe('AggregateRepository', () => {
-  describe('test', () => {
-    it('test', async () => {
-      expect(true).toBeTruthy();
+  let testAggregate: TestAggregate;
+  let testAggregateRepository: TestAggregateRepository;
+  let testEvent: TestEvent;
+  let testEventStore: EventStore;
+
+  beforeEach(() => {
+    testEvent = new TestEvent('id');
+    testAggregate = new TestAggregate();
+    testEventStore = new TestEventStore();
+    testAggregateRepository = new TestAggregateRepository(testEventStore);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('getById', () => {
+    it('should return TestAggregate', async () => {
+      const id = 'id';
+      const stream = `TestAggregate-${id}`;
+      const testEvents: Event[] = [testEvent];
+
+      jest.spyOn(testAggregateRepository, 'getNewInstance').mockReturnValueOnce(testAggregate);
+      jest.spyOn(testEventStore, 'read').mockResolvedValueOnce(testEvents);
+      jest.spyOn(testAggregate, 'applyEvent');
+
+      const aggregate = await testAggregateRepository.getById(id);
+
+      expect(testAggregateRepository.getNewInstance).toHaveBeenCalledTimes(1);
+      expect(testEventStore.read).toHaveBeenNthCalledWith(1, stream);
+      expect(testAggregate.applyEvent).toHaveBeenNthCalledWith(1, testEvent);
+      expect(aggregate.revision).toBeGreaterThan(-1);
+    });
+  });
+
+  describe('save', () => {
+    it('should handle save with applyEvent call', async () => {
+      const id = 'id';
+      testAggregate.id = id;
+      const stream = `TestAggregate-${id}`;
+
+      jest.spyOn(testEventStore, 'append');
+      jest.spyOn(testAggregate, 'resetChanges');
+
+      testAggregate.applyEvent(testEvent);
+      await testAggregateRepository.save(testAggregate);
+
+      expect(testEventStore.append).toHaveBeenNthCalledWith(1, stream, [], 0);
+      expect(testEventStore.append).toHaveBeenCalledTimes(1);
+    });
+
```

Comment:
Testing applyEvent and raiseEvent is not a part of `AggregateRepository`. `AggregateRepository` is only supposed to save **unsaved changes** which are tracked inside `Aggregate`. It does not matter how `changes` got populated in the `Aggregate`. 

Hope this makes sense. Feel free to reach out to me if you have any questions.

```suggestion
```